managed/src/main/java/com/yugabyte/yw/controllers/CustomerTaskController.java

```diff
@@ -104,21 +93,21 @@
     Map<UUID, List<CustomerTaskFormData>> taskListMap = new HashMap<>();
 
     for (CustomerTask task : customerTaskList) {
-      try {
-        CustomerTaskFormData taskData = new CustomerTaskFormData();
+      CustomerTaskFormData taskData = new CustomerTaskFormData();
 
-        JsonNode taskProgress = commissioner.getStatus(task.getTaskUUID());
-        // If the task progress API returns error, we will log it and not add that task
-        // to the task list for UI rendering.
-        if (taskProgress.has("error")) {
+      Optional<ObjectNode> taskProgress = commissioner.mayGetStatus(task.getTaskUUID());
+      // If the task progress API returns error, we will log it and not add that task
+      // to the task list for UI rendering.
+      if (taskProgress.isPresent()) {
+        if (taskProgress.get().has("error")) {
           LOG.error(
               "Error fetching Task Progress for "
                   + task.getTaskUUID()
                   + ", Error: "
-                  + taskProgress.get("error"));
+                  + taskProgress.get().get("error"));
         } else {
-          taskData.percentComplete = taskProgress.get("percent").asInt();
-          taskData.status = taskProgress.get("status").asText();
+          taskData.percentComplete = taskProgress.get().get("percent").asInt();
```

Comment:
You can extract method CustomerTaskFormData toTaskData(task, taskProgress) and burry try catch in there. Return null if any exception. If there is  non null return only then add it to the map.